### PR TITLE
The GridSearch tuner

### PR DESCRIPTION
**Pull request by haifeng-jin**
Added a GridSearch tuner.
It iterates all possible combinations in the search space, and supports conditional_scope.

### CODE REVIEW

File: keras_tuner/tuners/gridsearch.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+        This function is called on the second and later trials, but not on the
+        first one. Even new hps appear during search, the all hps in
+        self.get_space() should have corresponding values in the `values` arg.
+        This is because `values` is from the last trial, whose values has been
+        updated during that trial, which contains all seen hps.
+
+        We treat the default value of a hp as the first value, and the rest are
+        in normal order. We will enumerate the values according to this order.
+        Get all possible hyperparameter values
```
Comment (fchollet) [inline keras_tuner/tuners/gridsearch.py:108]: The description paragraphs above are unclear and have some typos, please clarify.

File: keras_tuner/tuners/gridsearch.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    combinations up to exhaustion.
+
+    For example:
+
+    ```py
+    optimizer = hp.Choice("model_name", values=["sgd", "adam"])
+    lr = hp.Choice("lr", values=[0.01, 0.1])
+    ```
+
+    This tuner will fit models for: ["sgd", 0.01], ["sgd", 0.1], ["adam", 0.01]
```
Comment (fchollet) [inline keras_tuner/tuners/gridsearch.py:158]: Use backticks

File: keras_tuner/tuners/gridsearch.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    """The Grid search tuner.
+
+    It will try all the possible hyperparameter
+    combinations up to exhaustion.
+
+    For example:
+
+    ```py
+    optimizer = hp.Choice("model_name", values=["sgd", "adam"])
+    lr = hp.Choice("lr", values=[0.01, 0.1])
```
Comment (fchollet) [inline keras_tuner/tuners/gridsearch.py:155]: Use `learning_rate`

File: keras_tuner/tuners/gridsearch.py [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    ```
+
+    This tuner will fit models for: ["sgd", 0.01], ["sgd", 0.1], ["adam", 0.01]
+    ["adam", 0.1].
+
+    For the following hyperparameter types, GridSearch will not exhaust all
+    possible values, but pick 10 samples evenly.
```
Comment (fchollet) [inline keras_tuner/tuners/gridsearch.py:162]: Why 10? It seems this should be configurable.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras_tuner/tuners/gridsearch.py (hunk 4 above)
Comment (haifeng-jin) [reply keras_tuner/tuners/gridsearch.py]: Revised the description for how to configure the granularity for Float and Int.

**Review by fchollet — APPROVED** [review]
Thanks for the updates! I made some copy edits. LGTM.

**Pull request merged**